nushell/reedline
# Fix vi mode change not pastable

While extending the tests for vi mode, I noticed that c$ (replace text until end of line with newly entered text) and cc (replace the entire line with newly entered text) don't actually cut the removed text. This fixed that. Verified with vim 9.1 that it should be cut a.k.a. pastable. 

Test:

setup:

1. $ cargo run --example demo -- --vi
2. enter a text (optionally use <Alt> + <Enter> to enter multiple line) 
3. press <esc> to enter vi normal mode and use the arrow keys to navigate somewhere in the line

actual test:

both "cc <esc> p" and "c$ <esc> p" should now leave the text unchanged by cutting out text and pasting it back again (, but moving the cursor to the end).
